cache.go

```diff
@@ -87,15 +90,37 @@ func (cd *cachedData) flush() {
 		return
 	}
 
-	file, err := os.OpenFile(cd.Path, os.O_RDWR|os.O_TRUNC, 0666)
+	file, err := fileutil.LockFile(cd.Path, os.O_RDWR, 0666)
 	if err != nil {
 		log.Printf("Cannot write to the cache file: %v", err)
```

Comment:
I don't think the user will even notice the blocking as it's very brief, but let's say we go with non-blocking. How should we proceed when the file is locked? Letting `zypper-docker` simply fail is unnecessary IMHO. We could do something like `"... file is locked, will try again in x seconds ..."`, but that's also a kind of blocking.

What do the others think?